src/scmrepo/git/__init__.py

```diff
@@ -327,12 +327,18 @@ def fetch_refspecs(
         progress: Optional[Callable[["GitProgressEvent"], None]] = None,
         **kwargs,
     ) -> typing.Mapping[str, SyncStatus]:
+        from urllib.parse import urlparse
+
         from .credentials import get_matching_helper_commands
 
-        if "dulwich" in kwargs.get("backends", self.backends.backends) and any(
-            get_matching_helper_commands(url, self.dulwich.repo.get_config_stack())
-        ):
-            kwargs["backends"] = ["dulwich"]
+        if "dulwich" in kwargs.get("backends", self.backends.backends):
+            credentials_helper = any(
+                get_matching_helper_commands(url, self.dulwich.repo.get_config_stack())
```

Comment:
IIRC credential-helper is not supposed to work for ssh, only for http urls. So this should be fixed in `get_matching_helper_commands()` or in upstream dulwich.

Regarding ssh not working in pygit2, let's investigate/fix it separately.

Reply:
> IIRC credential-helper is not supposed to work for ssh, only for http urls. So this should be fixed in get_matching_helper_commands() or in upstream dulwich.

Not sure I understand what you mean. The previous logic is preserved but the change is to also force `dulwich` if the URL is ssh

Reply:
`get_matching_helper_commands` should not match anything for ssh urls. Please try to address comments before merging.

Reply:
> get_matching_helper_commands should not match anything for ssh urls, so it should not enter this block.

Sorry, I am still not sure I follow. You mean it should be something like:

```python
if ssh or any(get_matching_helper_commands(url, self.dulwich.repo.get_config_stack()):
```